projects/ricochet-refresh/build

```diff
@@ -136,6 +136,7 @@ CMAKE_FLAGS="-DCMAKE_BUILD_TYPE=MinSizeRel
 [% END -%]
 
 mkdir build && cd build
+export SOURCE_DATE_EPOCH='[% c("timestamp") %]'
```

Comment:
Is there any reason to not do this step in the `var/set_default_env` script in rbm.conf so that it happens in all of the projects?

Reply:
On current main yes, there is at least one critical reason: `linuxdeploy` fails if SOURCE_DATE_EPOCH is set (see e.g. https://github.com/AppImage/AppImageKit/issues/929#issuecomment-926001098). However, adapting from a commit linked in the comment above, I was able to fix the `linuxdeploy` issue and to achieve fully reproducible Linux builds (i.e. also AppImages are reproducible). I am going to add those changes to this MR and if they are merged we can try setting SOURCE_DATE_EPOCH in `set/default_env` and see what happens with the rest of the build.

I don't know if there are other issues, but `grep -RI SOURCE_DATE_EPOCH projects`  says that `nsis` has an `export SOURCE_DATE_EPOCH=$(stat -c '%Y' Source)` and I don't know why it's needed.

Reply:
Ah it seems standard to export on the per-project level. I would reckon the `nsis` is leftover cruft from some years ago, as it does not appear in `tor-browser-build/main` so we can probably remove.

Reply:
See this patch from @PieroV :wave:
- https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/commit/018b7e26c08cc7e88a59e623aede1cf4d3353679